docs/DeveloperGuide.md

```diff
@@ -126,8 +126,20 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 This section describes some noteworthy details on how certain features are implemented.
 
 ### Data Structure: Tag
+[Tag](https://github.com/AY2021S1-CS2103T-F12-1/tp/blob/master/src/main/java/seedu/address/model/tag/Tag.java)
+allows a file to be tagged with a compulsory `TagName`, `FileAddress` and an optional `Label`. `TagName` must contain
+at least 1 word, alphanumeric and must be unique. `FileAddress` must contain a valid file path with the correct file 
+expression (i.e passing a file path like `C:\Windows\..` is valid for Windows and `./home/...` is valid for Linux. 
+`FileAddress` can take in a relative path or absolute path. User can make use of `FindCommand` by passing the 
+`TagName` as the keyword to find tagged files.  
+
 
 ### Data Structure: Label
+[Label](https://github.com/AY2021S1-CS2103T-F12-1/tp/blob/master/src/main/java/seedu/address/model/label/Label.java)
+allows `Tag ` to keep a `Label`. A `Label` must only contains alphanumeric. It is an optional field to `Tag`. 
```

Comment:
I think " Label allows 'Tag' to keep a 'Label' " might sounds a bit weird. But it's only personal opinion.

Reply:
Fix this!!